lib/middleware/karma.js

```diff
@@ -193,12 +194,12 @@ var createKarmaMiddleware = function (
 
             scriptUrls.push(filePath)
 
-            if (fileExt === '.css') {
+            if (fileExt === '.css' || fileType === 'css') {
               scriptTags.push(util.format(LINK_TAG_CSS, filePath))
               continue
             }
 
-            if (fileExt === '.html') {
+            if (fileExt === '.html' || fileType === 'html') {
               scriptTags.push(util.format(LINK_TAG_HTML, filePath))
               continue
             }
```

Comment:
On line 208, `scriptType` should be `fileType || SCRIPT_TYPE[fileExt] || 'text/javascript'`.  I think that would let us strip the code out of #2834 and just merge its tests.

Reply:
GitHub sadly wouldn't let me comment on the correct line.